Sources/MessagingInApp/Gist/Managers/MessageManager.swift

```diff
@@ -78,6 +97,25 @@ class MessageManager: EngineWebDelegate {
         }()
     }
 
+    /// Unsubscribes from InAppMessageState so that MessageManager can be deallocated.
+    func unsubscribeFromInAppMessageState() {
+        guard let subscriber = inAppMessageStoreSubscriber else { return }
+
+        logger.logWithModuleTag("Unsubscribing MessageManager from InAppMessageState", level: .debug)
+        inAppMessageManager.unsubscribe(subscriber: subscriber)
+        inAppMessageStoreSubscriber = nil
+    }
+
+    /// Removes EngineWebView from MessageManager and sets the delegate to nil to stop receiving JS events.
+    func removeEngineWebView() {
+        // If delegate is nil, then EngineWebView is already cleaned up.
+        guard let _ = engine.delegate else { return }
```

Comment:
this seems new and wasnt part of old code, any specific usecase?

Reply:
Yup, I added this to avoid calling `cleanEngineWeb` multiple times. To avoid any new explicit flag for this, I reused current delegate since we set it in `init`. Seems to work fine so far.